### DISCUSSION TITLE
Lectern should be able to set has_book directly

### DISCUSSION THREAD

**Discussion by TWME-TW** [PaperMC/Paper#10531] [Ideas]
### Is your feature request related to a problem?

I can set `has_book` to true directly through debugstick, but not through `org.bukkit.block.data.type.Lectern`. Obviously We need a method to set this value directly.

### Describe the solution you'd like.

Add a `setHasBook(boolean)`

### Describe alternatives you've considered.

Use Server.createBlockData(java.lang.String) to create a Lectern with a specific has_book value

**Comment by Lulu13022002** [comment]
Looks good however it's a little bit weird how those are missing since the impl is autogenerated by spigot. I will fix those in the generators branch for now but in the future stuff like this will be accessible through the property directly: https://github.com/PaperMC/Paper/pull/7223

**Comment by Lulu13022002** [comment]
Resolved in 1.21.5 with the new generator
